### PR TITLE
Add pagemenu to right side of Dash for R documentation

### PR DESCRIPTION
**Pull request by rpkyle**
This PR proposes to address #821 by
- loading the `dashUserGuideComponents` package for R
- adding the Pagemenu to the right hand side of the R docs pages
- introducing a draft of some of the callback logic to support backlinks and left hand side page navigation (currently disabled and untested); additional elements will be introduced in a later PR

Closes plotly/dash-docs#821.

### CODE REVIEW

File: run.R [hunk 1]
```diff
@@ -228,38 +229,75 @@ header <- htmlDiv(
...
+#               htmlDiv(id='chapter', className='content'),
+#               className='content-container'
+#             )
+#           ),
+#           pageMenu(id='pagemenu')
+#         )
+#       )
+#     )
+#   )
+# )
```
Comment (HammadTheOne) [inline run.R]: May want to remove this commented out codeblock before merging.

File: run.R [hunk 2]
```diff
@@ -228,38 +229,75 @@ header <- htmlDiv(
...
+        htmlDiv(id='backlinks-bottom', className='backlinks')
+        ),
+        className='rhs-content container-width'),
+            
+      pageMenu(id='pagemenu')
+      
+      ))
   )
 )
+)
```
Comment (HammadTheOne) [inline run.R]: A little bit of autoformat with RStudio on this code-block to clean it up:

```
app$layout(htmlDiv(
  list(
    # Stores used by examples.
    dccStore(id = 'memory'),
    dccStore(id = 'memory-output'),
    dccStore(id = 'local', storage_type = 'local'),
    dccStore(id = 'session', storage_type = 'session'),
    
    # div used in tests
    htmlDiv(id = 'wait-for-layout'),
    
    dccLocation(id = 'url', refresh = FALSE),
    
    header,
    
    htmlDiv(
      className = 'content-wrapper',
      children = list(
        #Sidebar(urls=SIDEBAR_INDEX),
        
        htmlDiv(list(
          htmlDiv(id = 'backlinks-top', className = 'backlinks'),
          htmlDiv(htmlDiv(id = 'chapter', className = 'content'),
                  className = 'content-container'),
          htmlDiv(id = 'backlinks-bottom', className = 'backlinks')
        ),
        className = 'rhs-content container-width'),
        
        pageMenu(id = 'pagemenu')
        
      )
    )
  )
))
```

File: run.R [hunk 3]
```diff
@@ -587,4 +625,73 @@ app$callback(
   }
 )
 
+app$callback(
```
Comment (HammadTheOne) [inline run.R]: Same as above, may want to add commented codeblocks to the scope of this PR or adjust it before merging.

File: run.R (hunk 1 above)
Comment (rpkyle) [reply run.R]: Yeah, I'll 🔪 these but deliberately commented them out so we could populate/refactor as needed once the other docs changes are done. I'll keep the branch alive after merging until that happens, then delete it.

File: run.R (hunk 2 above)
Comment (rpkyle) [reply run.R]: Thanks! 👔

Comment (rpkyle) [reply run.R]: fixed in 063d99c2

File: run.R (hunk 1 above)
Comment (rpkyle) [reply run.R]: Fixed in 063d99c2

File: dash_docs/assets/pagemenu.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+if(!window.dash_clientside) {window.dash_clientside = {};}
+window.dash_clientside.clientside = {
+  pagemenu: function (children) {
+    console.warn('updating pagemenu');
+    return '';
```
Comment (alexcjohnson) [inline dash_docs/assets/pagemenu.js]: Hmm I see this is just copied from the Python side, but the wildcards branch includes an optimization that if a callback returns the same value a prop already had, the update is skipped. I think that'll cause problems here, so better to do something variable like `return String(Date.now())`

cc @chriddyp - do you see any problems with preventing updates on unchanged props, other than needing to modify occasional hacks like this?

Comment (rpkyle) [reply dash_docs/assets/pagemenu.js]: Ah, good to know. I'll make this change. Thanks @alexcjohnson!

**Pull request merged**